### ISSUE TITLE
[Bug] GM_openInTab doesn't open as child with Tampermonkey

### ISSUE THREAD

**Issue by hanjo** [piroor/treestyletab#3272]
Hi,

I found a couple of issues from recent years describing the same problem with respect to Greasemonkey and noticed that TST is applying some workaround to mark a tab opened by GM_openInTab as a child of the current tab.

Some references: 
- https://github.com/piroor/treestyletab/issues/909
- https://github.com/piroor/treestyletab/issues/614
- https://github.com/piroor/treestyletab/issues/504
- https://github.com/piroor/treestyletab/issues/179

With Tampermonkey (v4.18.1) this doesn't seem to work on Firefox (v108.0.1), instead the tab is opened to the right of the current tab on the same level (with the current Tree Style Tab v3.9.11) - ignoring the container of the parent tab, but that's probably another issue.

Here is what I'm trying to execute:
```js
GM_openInTab('https://google.com', { loadInBackground: true, setParent: true })
```
(compare https://www.tampermonkey.net/documentation.php#api:GM_openInTab)

Any chance to have the workaround extended to Tampermonkey as well?

Thanks!

**Comment by piroor** [comment]
It is a Tempermonkey-side issue. TST respects `openerTabId` of opened tabs and TST cannot attach a new tab to the opener tab if there is no hint. Tempermonkey has to open a new tab with its opener information.

**Comment by hanjo** [comment]
Thanks for looking into this. I have opened a bug for Tampermonkey:
https://github.com/Tampermonkey/tampermonkey/issues/1678

**Comment by hanjo** [comment]
The issue has been fixed in Tampermonkey Beta 4.19.6178.
